Mathlib/Data/Real/EReal.lean

```diff
@@ -445,6 +445,144 @@ theorem eq_bot_iff_forall_lt (x : EReal) : x = ⊥ ↔ ∀ y : ℝ, x < (y : ERe
     exact ⟨x.toReal, coe_toReal_le h⟩
 #align ereal.eq_bot_iff_forall_lt EReal.eq_bot_iff_forall_lt
 
+/-! ### Intervals and coercion from reals -/
+
+lemma exists_between_ofReal {x z : EReal} (h : x < z) : ∃ y : ℝ, x < y ∧ y < z := by
+  obtain ⟨a, ha₁, ha₂⟩ := exists_between h
+  induction' a using EReal.rec with a₀
+  · exact (not_lt_bot ha₁).elim
+  · exact ⟨a₀, by exact_mod_cast ha₁, by exact_mod_cast ha₂⟩
+  · exact (not_top_lt ha₂).elim
+
+@[simp]
+lemma image_coe_Icc {x y : ℝ} : Real.toEReal '' Icc x y = Icc ↑x ↑y := by
+  refine (image_comp WithBot.some WithTop.some _).trans ?_
+  rw [WithTop.image_coe_Icc, WithBot.image_coe_Icc]
+  rfl
```

Comment:
Same here. Arguing with intersections should be shorter, although you will need to use the fact that the coercion is injective.

Reply:
This does not work; you end up with `Ico ↑x ⊤ ∩ Ioc ⊥ ↑y = Ici ↑x ∩ Iic ↑y`, but `Ioc ⊥ ↑y = Iic ↑y` (say) is false, so the equality of intersections is non-trivial.

Reply:
(Of course, one *can* prove the equality of the intersections, but then the proof is no longer simpler...)

Reply:
Ah yeah sorry, that's because you have more elements in the codomain.